src/sentry/static/sentry/app/actionCreators/tags.jsx

```diff
@@ -6,34 +6,78 @@ import AlertActions from 'app/actions/alertActions';
 
 const MAX_TAGS = 500;
 
-export function fetchTags(orgId, projectId = null) {
+function tagFetchSuccess(tags) {
+  const trimmedTags = tags.slice(0, MAX_TAGS);
+
+  if (tags.length > MAX_TAGS) {
+    AlertActions.addAlert({
+      message: t('You have too many unique tags and some have been truncated'),
+      type: 'warn',
+    });
+  }
+  TagActions.loadTagsSuccess(trimmedTags);
+}
+
+/**
+ * Fetch tags for a single project
+ */
+export function fetchProjectTags(orgId, projectId) {
   TagStore.reset();
   TagActions.loadTags();
   const api = new Client();
-  const url = projectId
-    ? `/projects/${orgId}/${projectId}/tags/`
-    : `/organizations/${orgId}/tags/`;
-
-  api.request(url, {
-    success: tags => {
-      const trimmedTags = tags.slice(0, MAX_TAGS);
-
-      if (tags.length > MAX_TAGS) {
-        AlertActions.addAlert({
-          message: t('You have too many unique tags and some have been truncated'),
-          type: 'warn',
-        });
-      }
-      TagActions.loadTagsSuccess(trimmedTags);
-    },
-    error: TagActions.loadTagsError,
+  const url = `/projects/${orgId}/${projectId}/tags/`;
+
+  const promise = api.requestPromise(url);
+  promise.then(tagFetchSuccess, TagActions.loadTagsError);
+
+  return promise;
+}
+
+/**
+ * Fetch tags for an organization or a subset or projects.
+ */
+export function fetchTags(api, orgId, projectIds = null) {
```

Comment:
How does this differ from `fetchProjectTags`?

Reply:
This uses the organization level endpoint which can fetch tag keys for multiple projects.

Reply:
Is it possible to use this to search tags for a single project as well as multiple projects?

Reply:
@markstory 	^

Reply:
We could, but the permissions for organization endpoints and project endpoints are a bit different right now. Once we've normalized permissions we could stop using the project endpoints.